cmd/update-main.go

```diff
@@ -68,90 +68,108 @@ const (
 	minioReleaseURL           = "https://dl.minio.io/server/minio/release/" + runtime.GOOS + "-" + runtime.GOARCH + "/"
 )
 
-func getCurrentReleaseTime(minioVersion, minioBinaryPath string) (releaseTime time.Time, err error) {
-	if releaseTime, err = time.Parse(time.RFC3339, minioVersion); err == nil {
+// updateCheckParams - collection of parameters used to check if the
+// version of the current server process is older than the latest
+// available official release. They enable the testability of various
+// routines in the update checking process.
+type updateCheckParams struct {
+	// version of the current server process
+	version string
+	// path to the server binary
+	serverBinaryPath string
+	// file on the server to check for docker env
+	cgroupFile string
+	// URL of release checksum file (contains version of latest
+	// release)
+	releaseChecksumURL string
+	// current process run mode
+	serverMode string
+}
+
+// getCurrentReleaseTime - returns this process's release time.  If it
+// is official minio version, parsed version is returned else minio
+// binary's mod time is returned.
+func (u *updateCheckParams) getCurrentReleaseTime() (releaseTime time.Time,
```

Comment:
Another official resource on this topic is here - https://golang.org/doc/faq#methods_on_values_or_pointers - appears that value receivers are good (i.e. efficient and clear) for small structs according to the doc.

Also, in some cases, for value receivers, a copy of the struct may end up getting allocated on the heap rather than the stack. This creates more work for the GC as items allocated on the function's call stack are freed at the end of a function call (and don't need the GC). Go determines where to put the allocation based on [Escape Analysis](https://en.wikipedia.org/wiki/Escape_analysis). I couldn't find any official docs about the implementation but there are references to it in golang issue tracker, reddit, mailing list and other places.